minio/minio
# re-write the KMS get started guide

## Description
This commit updates the KMS getting started guide
and replaces the legacy MinIO<-->Vault setup with a
MinIO<-->KES<-->Vault setup.

Therefore, add some architecture ASCII diagrams and
provide a step-by-step guide to setup Vault, KES and
MinIO such that MinIO can encrypt objects with KES +
Vault.

The legacy Vault guide has been moved to `./vault-legacy.md`.

## Motivation and Context
KMS , KES, Vault

## How to test this PR?
Follow the guide and verify that you can setup Vault + KES + MinIO successfully.

## Types of changes

## Checklist:
